application-licensing-licensor/application-licensing-licensor-api/src/test/java/com/xwiki/licensing/internal/DefaultLicensingConfigurationTest.java

```diff
@@ -75,4 +96,22 @@ public void getBlocklistWithEmptyList() throws Exception
 
         assertEquals(Collections.emptyList(), mocker.getComponentUnderTest().getAutoUpgradeBlocklist());
     }
+
+    @Test
+    public void getLocalStorePath() throws Exception
+    {
+        when(this.configurationSource.getProperty("licensing.localStorePath")).thenReturn("path");
+        this.mocker.getComponentUnderTest().getLocalStorePath();
+
+        verify(this.environment, never()).getPermanentDirectory();
+    }
+
+    @Test
+    public void getLocalStorePathWithNullProperty() throws Exception
+    {
+        when(this.configurationSource.getProperty("licensing.localStorePath")).thenReturn(null);
+        this.mocker.getComponentUnderTest().getLocalStorePath();
+
+        verify(this.environment).getPermanentDirectory();
```

Comment:
Why not use ``when`` to return some value and thus make sure that the permanent directory is actually used (the return value might as well be discarded).